### ISSUE TITLE
work with the room architecture component

### ISSUE THREAD

**Issue by wuyang102** [rharter/auto-value-gson#187]
The room architecture component has support auto value now. But it can not work this library as it will generate a medium class like $AutoValue_xxx.java

**Comment by JakeWharton** [comment]
Isn't this a Room bug then?

**Comment by wuyang102** [comment]
> Isn't this a Room bug then?

No. The abstract medium class start with $ symbol is created by this library.

**Comment by JakeWharton** [comment]
_Any extension_ will create that type (and potentially more) and it's
perfectly valid since it sits between the `AutoValue_`-prefixed type and
the user-written type. Please file a Room bug.
